portality/lib/edges.py

```diff
@@ -27,19 +27,15 @@ def query(self):
             for term in self.terms:
                 musts.append({"terms" : term})
 
-        query = {"match_all" : {}}
         if self.query_string is not None:
-            query = {"query_string" : {"default_operator" : "AND", "query" : self.query_string}}
+            qs = {"query_string" : {"default_operator" : "AND", "query" : self.query_string}}
+            musts.append(qs)
 
+        query = {"match_all": {}}
         if len(musts) > 0:
             query = {
                 "bool" : {
-                    "filter" : {
-                        "bool": {
-                            "must": musts
-                        }
-                    },
-                    "must" : query
+                    "must": musts
```

Comment:
I cherry picked this change as a hotfix, which as since been modified, so this change will need to be resolved when we merge everything down to develop eventually